src/app.jsx

```diff
@@ -357,3 +363,124 @@ export const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }) => {
         </Select>
     );
 };
+
+const DropdownWithKebab = ({ selected, path }) => {
+    const Dialogs = useDialogs();
+    const [isOpen, setIsOpen] = useState(false);
+
+    const onToggleClick = () => {
+        setIsOpen(!isOpen);
+    };
+    const onSelect = (_event, itemId) => {
+        setIsOpen(false);
+    };
+
+    const deleteItem = () => {
+        const itemPath = "/" + path.join("/") + "/" + selected.name;
+        Dialogs.show(<ConfirmDeletionDialog selected={selected} itemPath={itemPath} />);
+    };
+
+    return (
+        <Dropdown
+            isPlain
+            isOpen={isOpen}
+            onSelect={onSelect}
+            onOpenChange={isOpen => setIsOpen(isOpen)}
+            popperProps={{ position: "right" }}
+            toggle={toggleRef =>
+                <MenuToggle ref={toggleRef} variant="plain" onClick={onToggleClick} isExpanded={isOpen}>
+                    <EllipsisVIcon />
+                </MenuToggle>}
+        >
+            <DropdownList>
+                <DropdownItem itemId='delete-item' key="delete-item" isDisabled={!selected} onClick={deleteItem} className={!selected ? "" : "pf-m-danger"}>
+                    {_("Delete")}
+                </DropdownItem>
+            </DropdownList>
+        </Dropdown>
+    );
+};
+
+const ConfirmDeletionDialog = ({ selected, itemPath }) => {
+    const Dialogs = useDialogs();
+
+    const deleteItem = () => {
+        if (selected.type === "file") {
+            cockpit.spawn(["rm", itemPath]).then(Dialogs.close)
```

Comment:
When you chain more than one promises, please new line for each chain part. In this case you could though reuse the then second argument.

```
cockpit.spawn(["rm", itemPath])
.then(Dialogs.close, () => Dialogs.show(<ForceDeleteModal itemPath={itemPath} />) )
```